maestro/audit.py

```diff
@@ -351,6 +358,57 @@ def from_config(cfg):
             cfg.get('timeout', WebHookAuditor.DEFAULT_TIMEOUT))
 
 
+class ExecuteScriptAuditor(BaseAuditor):
+    """Auditor that executes scripts with maestro action information."""
+
+    def __init__(self, script, args, level):
+        super(ExecuteScriptAuditor, self).__init__(level)
+        if not script:
+            raise exceptions.InvalidAuditorConfigurationException(
+                'Missing script to execute!')
+
+        self._script = script
+        self._args = args
+
+    def _format_container_dict(self, what):
+        if isinstance(what, entities.Entity):
+            return None
+        d = {}
+        for item in what:
+            x = dict(ship=item.ship.ip, service=item.service.name)
+            d[item.name] = x
+        return StringIO(json.dumps(d)).read()
+
+    def action(self, level, what, action, who):
+        if not self._should_audit(level):
+            return
+        ships = self._format_container_dict(what)
+
+        if not ships:
+            return
+
+        args_ls = self._args.format(action=action,
+                                    what=what, who=who).split()
+        cmd = [self._script]
+        cmd.extend(args_ls)
+        from subprocess import Popen, PIPE
+        p = Popen(cmd, stdout=PIPE, stdin=PIPE, stderr=PIPE)
+        p.communicate(input=ships)
```

Comment:
This doesn't do any error checking. You should just use `subprocess.check_call()`. I also don't think it's relevant to pipe standard out, since the Maestro output is fairly controlled.